cmd/bootstrap/run/execution_state.go

```diff
@@ -67,11 +54,38 @@ func GenerateExecutionState(
 		<-compactor.Done()
 	}()
 
-	return bootstrap.NewBootstrapper(
-		zerolog.Nop()).BootstrapLedger(
-		ledgerStorage,
-		accountKey,
-		chain,
-		bootstrapOptions...,
-	)
+	stateCommitment, err := bootstrap.
+		NewBootstrapper(zerolog.Nop()).
+		BootstrapLedger(
+			ledgerStorage,
+			accountKey,
+			chain,
+			bootstrapOptions...,
+		)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+
+	checkpointer, err := ledgerStorage.Checkpointer()
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+
+	root := 0
+	err = checkpointer.Checkpoint(root)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+
+	trie, err := checkpointer.LoadCheckpoint(root)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
```

Comment:
I would suggest we do this instead:
```

tries := ledgerStorage.Tries()
trie := findTrieByStateCommit(tries, stateCommitment)

```

We could implement the `findTrieByStateCommit` in the `ledger.FindTrieByStateCommit`

Reply:
so avoid using checkpointer? makes sense, I didn't like hardcoding root at 0 anyway.

Reply:
Yeah, but more importantly to validate the statecommitment of the trie.